src/shims/foreign_items.rs

```diff
@@ -219,22 +215,23 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             .first_attr_value_str_by_name(&attrs, sym::link_name)
             .unwrap_or_else(|| this.tcx.item_name(def_id));
         let link_name = link_name_sym.as_str();
```

Comment:
Now we can make things consistent by renaming `link_name_sym` to `link_name`, and just calling `as_str` on it where needed. :)